### DISCUSSION TITLE
Issues when setting the size for floating windows.

### DISCUSSION THREAD

**Discussion by Samueru-sama** [i3/i3#5775] [Q&A]
I'm trying to make floating windows a certain size, for that I just added this to my i3 config: 

for_window [floating] resize set 680 420

And it works the first time that I make a tiled window float, however if I then tile the window and float it again the effects are no longer applied and the floating window takes the entire screen space again. 

Then I thought about making the resize command when I use my keybind toggle to float, like this: 

bindsym $mod+Shift+space floating toggle, resize set 680 420

The issue is that when I toggle the window back tiling the resize command is also invoked changing the size of the tiled window. 

So how can I make i3 set only floating windows a certain size?

**Comment by FredySandoval** [comment]
Modify the keybinding to use a semicolon (;) to separate commands. This allows you to apply the resize only when the window is floating:


```
# Floating windows size and position
for_window [floating] resize set 680 420
for_window [floating] move position center

# Toggle floating with size reset
bindsym $mod+Shift+space floating toggle; [floating] resize set 680 420, move position center

# Optional: Reset size for specific applications when they float
# for_window [class="Specific_App" floating] resize set 680 420
```

**Reply by Samueru-sama** [reply]
Thanks for the reply, this is actually what I've been [using](https://github.com/Samueru-sama/dotfiles/blob/main/Local/config/i3/config#L219) but it has the problem that all the floating windows have the same size when set to floating. 

For example I have zenity to be `290 240` but if I turn it into tiled window and then black to floating, it gets stuck being a `708 385` window. 

I think I will end up writting an script that gets the current window class and uses i3-msg to always force certain floating windows to have X size.
